openapiart/common.go

```diff
@@ -187,6 +191,54 @@ func (api *api) under_review(message string) {
 	fmt.Printf("warning: %s\n", message)
 }
 
+func (api *api) FromError(err error) (Error, bool) {
+	if rErr, ok := err.(Error); ok {
+		return rErr, true
+	}
+
+	rErr := NewError()
+	if err := rErr.FromJson(err.Error()); err == nil {
+		return rErr, true
+	}
+
+	return api.fromGrpcError(err)
+}
+
+func (api *api) setResponseErr(obj Error, code int32, message string) {
+	errors := []string{}
+	errors = append(errors, message)
+	obj.Msg().Code = code
+	obj.Msg().Errors = errors
+}
+
+func (api *api) fromGrpcError(err error) (Error, bool) {
+	st, ok := status.FromError(err)
+	if ok {
+		rErr := NewError()
+		if err := rErr.FromJson(st.Message()); err == nil {
+			rErr.Msg().Code = int32(st.Code())
+			return rErr, true
+		}
+
+		api.setResponseErr(rErr, int32(st.Code()), st.Message())
+		return rErr, true
+	}
+
+	return nil, false
```

Comment:
what if ok is false, I read through the documentation and it says that it returns ok as false if the error is not supported with the package status, now if there is an error which is not supported the package it still gives us st just that ok is false, according to me if ok is false we should still be returning an error otherwise we will miss it